t/03.scripts/tweak_pipeline_output.t

```diff
@@ -0,0 +1,114 @@
+
+#Check pipeline (show, set, error)
+push @tweak_requests,  ["-tweak" => "pipeline.hive_use_param_stack=20", "-json"];
+push @tweak_requests,  ["-tweak" => "pipeline.hive_use_param_stack?", "-json"];
+push @tweak_requests,  ["-tweak" => "pipeline.hive_use_param_stac?", "-json"];
+
+#Check analysis (show, delete, set) / (resource_class / is_excluded / error)
+push @tweak_requests,  ["-tweak" => "analysis[perform_cmd].resource_class?", "-json"];
+push @tweak_requests,  ["-tweak" => "analysis[perform_cmd].resource_class=20", "-json"];
+push @tweak_requests,  ["-tweak" => "analysis[perform_cmd].resource_class#", "-json"];
+
+push @tweak_requests,  ["-tweak" => "analysis[perform_cmd].is_excluded?", "-json"];
+push @tweak_requests,  ["-tweak" => "analysis[perform_cmd].is_excluded=20", "-json"];
+push @tweak_requests,  ["-tweak" => "analysis[perform_cmd].is_excluded#", "-json"];
+
+push @tweak_requests,  ["-tweak" => "analysis[perform_cmd].some_wrong_attr?", "-json"];
+push @tweak_requests,  ["-tweak" => "analysis[perform_cmd].some_wrong_attr=20", "-json"];
+push @tweak_requests,  ["-tweak" => "analysis[perform_cmd].some_wrong_attr#", "-json"];
+
+#Check analysis.wait_for (show, delete, set)
+push @tweak_requests,  ["-tweak" => "analysis[perform_cmd].wait_for=perform_cmd", "-json"];
+push @tweak_requests,  ["-tweak" => "analysis[perform_cmd].wait_for?", "-json"];
+push @tweak_requests,  ["-tweak" => "analysis[perform_cmd].wait_for#", "-json"];
+
+#Check analysis.flow_into (show, delete, set)
+push @tweak_requests,  ["-tweak" => "analysis[perform_cmd].flow_into=perform_cmd", "-json"];
+push @tweak_requests,  ["-tweak" => "analysis[perform_cmd].flow_into?", "-json"];
+push @tweak_requests,  ["-tweak" => "analysis[perform_cmd].flow_into#", "-json"];
+
+#Check analysis.param (show, delete, set)
+push @tweak_requests,  ["-tweak" => "analysis[perform_cmd].param[base]=10", "-json"];
+push @tweak_requests,  ["-tweak" => "analysis[perform_cmd].param[base]?", "-json"];
+push @tweak_requests,  ["-tweak" => "analysis[perform_cmd].param[base]#", "-json"];
+
+#Check resource_class (show, set, error)
+push @tweak_requests,  ["-tweak" => "resource_class[urgent].LSF=-q yesteryear", "-json"];
+push @tweak_requests,  ["-tweak" => "resource_class[urgent].LSF?", "-json"];
+
+foreach my $request (@tweak_requests) {
+  my $stdout = capture_stdout {
+    tweak_pipeline($pipeline_url, $request);
+  };
+  is_valid_json $stdout;
+  my $stdoutJson = decode_json($stdout);
+  use Data::Dumper;
+  ok(scalar @{$stdoutJson->{Tweaks}} > 0, "Tweaks responce recieved for " . join (' ', @{$request}) . Dumper($stdoutJson->{Tweaks}));
+  foreach my $tweakJson (@{$stdoutJson->{Tweaks}}) {
+    ok($tweakJson->{Object}->{Type} eq "Pipeline" ||
+      $tweakJson->{Object}->{Type} eq "Analysis" ||
+      $tweakJson->{Object}->{Type} eq "Resource class",
+      'Object type is correct for ' . join (' ', @{$request}));
+
+    ok( $tweakJson->{Error} ||
```

Comment:
So basically if there is an error, nothing in the JSON can be relied upon apart from Object>Type. I think we can guarantee more than that. I would say that the output structure should always have `Action` and `Object`, and then either `Return` or `Error`. This means going back to HivePipeline.pm and adding the missing fields whenever there is an error.

Reply:
If we put "delete" action - and it's not expected, we can't put it as action: "delete" error: ..., that would mean delete error, but the error is about wrong parameter. The same about object - it would mean the error is about attempt on the object - it can't be deleted, but the error is just about this is forbidden parameters combination

Reply:
There are some places where we can tell that the action was - "DELETE" and in that places we now fill the action, but there are also places (Failed to parcel tweak at line 963) where we can't fill the action - that's why we can't guarantee action to be there.

Reply:
OK, so the JSON structure and the test should follow this reasoning:
1. Can we parse the tweak ?
   * NO: output is a JSON with _only_ `error`
   * YES: we populate the JSON with `Object` and `Action`. Then, 2) can we execute the action ?
      * NO: add `error`
      * YES: add `Return`